Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -1026,7 +1027,21 @@ extension PackageInfo {
             settingsDictionary["SWIFT_VERSION"] = .string(swiftLanguageVersion)
         }
 
-        return settingsDictionary.isEmpty ? nil : .settings(base: settingsDictionary)
+        let hasCustomConfig = buildConfigs.firstIndex(where: { !["Debug", "Release"].contains($0.name) }) != nil
```

Comment:
It would probably be better to make this field optional. If user populates it then it has a value, otherwise it's nil and we use defaults.

Reply:
I think the logic should be that we always use what we got. If the user wants to use the defaults it' already handled by the init AFAIR